stage0/src/asm/boot.s

```diff
@@ -65,11 +65,27 @@ _protected_mode_start:
     mov $0xc0010131, %ecx     # SEV_STATUS MSR. See Section 15.34.10 in AMD64 Architecture Programmer's
                               # Manual, Volume 2 for more details.
     rdmsr                     # EDX:EAX <- MSR[ECX]
-    and $3, %eax              # eax &= 0b11;
+    and $0b111, %eax          # eax &= 0b111;
                               # Bit 0 - SEV enabled
                               # Bit 1 - SEV-ES enabled
+                              # Bit 2 - SEV-SNP active
     test %eax, %eax           # is it zero?
     je no_encryption          # if yes, jump to no_encryption
+    # See if we're under SEV-SNP, and if yes, pre-emptively PVALIDATE the first megabyte of memory, as that's
+    # where we'll be storing many data structures.
+    and $0b100, %eax          # eax &= 0b100; -- SEV-SNP active
+    test %eax, %eax           # is it zero?
```

Comment:
What does it mean if "it is zero?"

Reply:
`if eax == 0`, this is the way how you check if a register is zero in x86 assembly.